src/motor_hardware.cc

```diff
@@ -863,53 +864,42 @@ void MotorDiagnostics::firmware_options_status(DiagnosticStatusWrapper &stat) {
 // The I2C address is the 8-bit address which is the 7-bit addr shifted left in some code
 // If chipRegAddr is greater than 1 we write this out for the internal chip address for the following read(s)
 //
-// Returns number of bytes read where 0 or less implies some form of failure
-//
 // NOTE: The i2c8bitAddr will be shifted right one bit to use as 7-bit I2C addr
 //
 static int i2c_BufferRead(const char *i2cDevFile, uint8_t i2c8bitAddr, 
-                          uint8_t *pBuffer, int16_t chipRegAddr, uint16_t NumByteToRead)
+                          uint8_t *pBuffer, uint16_t NumBytesToRead)
 {
-   int bytesRead = 0;
-   int retCode   = 0;
-
-    int fd;                                         // File descrition
-    int  address   = i2c8bitAddr >> 1;              // Address of the I2C device
-    uint8_t buf[8];                                 // Buffer for data being written to the i2c device
+    int fd;                                         // File descriptor
+    int retCode = 0;
+    int byteRead = 0;
+    int slaveAddress = i2c8bitAddr >> 1;            // Address of the I2C device
 
     if ((fd = open(i2cDevFile, O_RDWR)) < 0) {      // Open port for reading and writing
```

Comment:
Can you please also test this line 877 like that: change O_RDWR into O_RDONLY 
Because:
- O_RDWR is opening file for reading and writing
- O_RDONLY is opening file for reading

Reply:
I can test to verify it works in the only mode we run it in now which is read one byte.
BUT you seem to have not changed it so there would be no value to retest.   
We should check the same thing that would be merged later and not let me mess with it on my own code copy.